lib/middleware/route.js

```diff
@@ -69,13 +69,18 @@ exports.middleware = function route(next, app) {
     function addRoute(method, path, fn, name) {
         var keys = [];
         var spec = {keys: keys, fn: fn};
-        spec.pattern = path instanceof RegExp ? path : normalizePath(path, keys);
-        spec.weight = calcWeight(path);
+        if (path instanceof RegExp) {
```

Comment:
Adding `RegExp` as path spec breaks the reverse lookup. Is there a special reason for adding this? I don't see a use in the locale middleware for this.